src/planner/linux.rs

```diff
@@ -1,21 +1,31 @@
+use std::{collections::HashMap, path::Path};
+
+use tokio::process::Command;
+use which::which;
+
+use super::ShellProfileLocations;
+#[cfg(all(target_os = "macos", not(feature = "nix-community")))]
```

Comment:
Seems like making this macos only in the linux.rs file might be a mistake?
```suggestion
#[cfg(not(feature = "nix-community"))]
```